kinto/core/authorization.py

```diff
@@ -283,4 +297,13 @@ def _find_required_permission(self, request, service):
                 else:
                     required_permission = "write"
 
+        # In the case of a "POST" on a plural endpoint, if an "id" was
+        # specified, then the object is returned. The required permission
+        # is thus "read" on this object.
+        if request.method.lower() == "post" and self.current_object is not None:
+            permission_object_id = self.get_permission_object_id(
+                request, object_id=self._resource.object_id
+            )
+            required_permission = "read"
```

Comment:
Wait a sec. Won't the POST in the linked issue replace the data? Shouldn't it therefore be `write`? I guess if they can't do the POST then we want `read` to show them the existing record?

Reply:
No, if `id` is specified and object exists, then we return the existing object:
https://docs.kinto-storage.org/en/stable/api/1.x/records.html#uploading-a-record

Reply:
I see. OK, sure.